decidim-regulations/app/controllers/decidim/regulations/regulation_controller.rb

```diff
@@ -25,84 +23,78 @@ def show
-        @current_participatory_process ||= collection.find_by(slug: params[:slug])
+        @current_participatory_space ||= organization_participatory_processes.where(slug: params["slug"]).or(
+          organization_participatory_processes.where(id: params["slug"])
+        ).first!
       end
 
-      def stats
-        @stats ||= Decidim::ParticipatoryProcesses::ParticipatoryProcessStatsPresenter.new(participatory_process: current_participatory_process)
+      def published_processes
+        @published_processes ||= Decidim::ParticipatoryProcesses::OrganizationPublishedParticipatoryProcesses.new(current_organization, current_user).query.where(decidim_participatory_process_group_id: Rails.application.config.regulation)
       end
 
-      def filter
-        @filter = params[:filter] || default_filter
+      def promoted_participatory_processes
+        @promoted_participatory_processes ||= published_processes.promoted
       end
 
-      def regulation
-        @regulation = request.original_fullpath.include? "/regulations"
+      # This is customized because GENCAT don't Processes Groups on Index Page
+      def collection
+        @collection ||= participatory_processes
       end
+      # This is customized because GENCAT don't Processes Groups on Index Page
 
-      def default_filter
-        "active"
+      def filtered_processes
+        search.results.where(decidim_participatory_process_group_id: Rails.application.config.regulation)
       end
 
-      def types
-        @types ||= ::DecidimType.all
+      # This is customized because GENCAT
+      def participatory_processes
+        @participatory_processes ||= filtered_processes
       end
+      # This is customized because GENCAT
 
-      def departments
-        @departments ||= ::DecidimDepartment.all
+      def participatory_process_groups
+        @participatory_process_groups ||= Decidim::ParticipatoryProcessGroup
+                                          .where(id: filtered_processes.grouped.group_ids)
       end
 
-      def filter_type
-        if params.has_key?(:type_id)
-          @filter_type = params[:type_id]
-        else
-          @filter_type = 0
-        end
+      def stats
+        @stats ||= ParticipatoryProcessStatsPresenter.new(participatory_process: current_participatory_space)
       end
 
-      def theme_type
-        if params.has_key?(:theme_id)
-          @department_type = params[:theme_id]
-        else
-          @department_type = 0
-        end
+      def metrics
+        @metrics ||= ParticipatoryProcessMetricChartsPresenter.new(participatory_process: current_participatory_space)
       end
 
-      def status
-        if params.has_key?(:status)
-          @status = params[:status]
-        else
-          @status = 2
-        end
+      def default_date_filter
+        return "opened" if published_processes.where("start_date < ? AND end_date > date(?)", Date.current, Date.current).order(start_date: :desc).any?(&:active?)
```

Comment:
what do you want to achieve here?

I think it would be enough to call: `published_processes.any?(&:active?)`

```ruby
    def active?
      return false if start_date.blank?
      start_date < Date.current && (end_date.blank? || end_date > Date.current)
    end
```

Reply:
You're right! Fixed